watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack.py

```diff
@@ -0,0 +1,326 @@
+
+    # Scaling
+    m.fs.properties.set_default_scaling("flow_mol_phase_comp", 1, index=("Liq", "H2O"))
+    m.fs.properties.set_default_scaling(
+        "flow_mol_phase_comp", 1e1, index=("Liq", "Na_+")
+    )
+    m.fs.properties.set_default_scaling(
+        "flow_mol_phase_comp", 1e1, index=("Liq", "Cl_-")
+    )
+    iscale.set_scaling_factor(m.fs.EDstack.cell_width, 10)
+    iscale.set_scaling_factor(m.fs.EDstack.cell_length, 10)
+    iscale.calculate_scaling_factors(m)
+    return m
+
+
+def set_operating_conditions(m):
+
+    solver = get_solver()
+
+    # ---specifications---
+    # Fix state variables at the origin
+    m.fs.feed.properties[0].pressure.fix(101325)  # feed pressure [Pa]
+    m.fs.feed.properties[0].temperature.fix(273.15 + 25)  # feed temperature [K]
+    m.fs.feed.properties[0].flow_mol_phase_comp["Liq", "H2O"].fix(4.8)
+    m.fs.feed.properties[0].flow_mol_phase_comp["Liq", "Na_+"].fix(1.476e-2)
+    m.fs.feed.properties[0].flow_mol_phase_comp["Liq", "Cl_-"].fix(1.476e-2)
+
+    # Fix separator's split_fraction to 0.5, i.e., equal flows into the diluate and concentrate channels
+    m.fs.separator.split_fraction[0, "inlet_diluate"].fix(0.5)
+
+    # Fix ED unit vars
+    m.fs.EDstack.water_trans_number_membrane["cem"].fix(5.8)
+    m.fs.EDstack.water_trans_number_membrane["aem"].fix(4.3)
+    m.fs.EDstack.water_permeability_membrane["cem"].fix(2.16e-14)
+    m.fs.EDstack.water_permeability_membrane["aem"].fix(1.75e-14)
+    m.fs.EDstack.voltage_applied.fix(5)
+    m.fs.EDstack.electrodes_resistance.fix(0)
+    m.fs.EDstack.cell_pair_num.fix(100)
+    m.fs.EDstack.current_utilization.fix(1)
+    m.fs.EDstack.spacer_thickness.fix(2.7e-4)
+    m.fs.EDstack.membrane_areal_resistance["cem"].fix(1.89e-4)
+    m.fs.EDstack.membrane_areal_resistance["aem"].fix(1.77e-4)
+    m.fs.EDstack.cell_width.fix(0.1)
+    m.fs.EDstack.cell_length.fix(0.79)
+    m.fs.EDstack.membrane_thickness["aem"].fix(1.3e-4)
+    m.fs.EDstack.membrane_thickness["cem"].fix(1.3e-4)
+    m.fs.EDstack.solute_diffusivity_membrane["cem", "Na_+"].fix(1.8e-10)
+    m.fs.EDstack.solute_diffusivity_membrane["aem", "Na_+"].fix(1.25e-10)
+    m.fs.EDstack.solute_diffusivity_membrane["cem", "Cl_-"].fix(1.8e-10)
+    m.fs.EDstack.solute_diffusivity_membrane["aem", "Cl_-"].fix(1.25e-10)
+    m.fs.EDstack.ion_trans_number_membrane["cem", "Na_+"].fix(1)
+    m.fs.EDstack.ion_trans_number_membrane["aem", "Na_+"].fix(0)
+    m.fs.EDstack.ion_trans_number_membrane["cem", "Cl_-"].fix(0)
+    m.fs.EDstack.ion_trans_number_membrane["aem", "Cl_-"].fix(1)
+
+    # check degrees of freedom
+    if degrees_of_freedom(m) != 0:
+        raise RuntimeError(
+            "The set_operating_conditions function resulted in {} "
+            "degrees of freedom rather than 0. This error suggests "
+            "that too many or not enough variables are fixed for a "
+            "simulation.".format(degrees_of_freedom(m))
+        )
```

Comment:
This could be replaced with our utility function: watertap.core.util.initilization.check_dof

We should probably update this for all flowsheets to centralize this check and error message